src/ds/d400/d400-factory.cpp

```diff
@@ -649,6 +649,27 @@ namespace librealsense
 
             return tags;
         };
+
+        void hardware_reset() override 
+        {
+            d400_device::hardware_reset();
+            auto dev_info = this->get_device_info();
+            auto non_const_device_info = std::const_pointer_cast< librealsense::device_info >( dev_info );
+            std::vector< std::shared_ptr< device_info > > devices{ non_const_device_info };
+            auto ctx = get_context();
+            //creating fake notification to trigger invoke_devices_changed_callbacks, causing connection and disconnection of mipi device
+            std::thread fake_notification(
+                //sending a copy of ctx since the ctx of the device will be destroyed
+                [ ctx, devices ]() {
+                    ctx->invoke_devices_changed_callbacks(
+                        devices, {} );
+                    std::this_thread::sleep_for( std::chrono::milliseconds( 3000 ) ); 
```

Comment:
Please add a comment before this line explaining why sleep and why 3 sec

Reply:
Done